src/components/FeatureViews/Courses/CourseManagement.js

```diff
@@ -216,65 +205,89 @@ const CourseDisplayCard = ({
 
 const CourseManagement = () => {
   const classes = useStyles();
-  const [tab1, setTab1] = useState("");
-  const [tab2, setTab2] = useState("");
-  const [tab3, setTab3] = useState("");
-  const [tab4, setTab4] = useState("");
+  const [sortByDate, setSortByDate] = useState("");
+  const [filterByGrades, setFilterByGrades] = useState("");
+  const [filterBySubjects, setFilterBySubjects] = useState("");
+  const [filterByInstructors, setFilterByInstructors] = useState("");
 
-  const handleChange1 = (event) => {
-    setTab1(event.target.value);
-  };
-
-  const handleChange2 = (event) => {
-    setTab2(event.target.value);
-  };
-
-  const handleChange3 = (event) => {
-    setTab3(event.target.value);
-  };
-
-  const handleChange4 = (event) => {
-    setTab4(event.target.value);
+  const handleChange = (setTab) => (event) => {
+    setTab(event.target.value);
   };
 
   const GET_COURSES = gql`
-  query getCourses {
-    courses {
-      dayOfWeek
-      endDate
-      endTime
-      title
-      startTime
-      startDate
-      instructor {
-        user {
-          firstName
-          lastName
+    query getCourses {
+      courses {
+        dayOfWeek
+        endDate
+        endTime
+        title
+        startTime
+        academicLevel
+        startDate
+        instructor {
+          user {
+            firstName
+            lastName
+            id
+          }
         }
+        courseCategory {
+          id
+          name
+        }
+        courseId
       }
-      courseId
     }
-  }
   `;
 
   const { data, loading, error } = useQuery(GET_COURSES);
 
   if (loading) return <Loading />;
   if (error) return console.error(error.message);
-  // console.log(data)
+  console.log(data.courses);
 
-  const courseDisplayList = data.courses.map((courses) => (
-    <CourseDisplayCard
-      title={courses.title}
-      day={courses.dayOfWeek}
-      endDate={courses.endDate}
-      endTime={courses.endTime}
-      startTime={courses.startTime}
-      startDate={courses.startDate}
-      instructor={courses.instructor}
-      id={courses.courseId}
-    />
-  ));
+  const subjectList = data.courses.reduce(
```

Comment:
I did this a little differently and would like to get your thoughts on this.

Reply:
hmm so what is filterCriteria supposed to be in this case? when i wrote up my example, i thought it would be a function that takes in 2 variables that defines how you're supposed to filter it (similar to the .filter() function for js array)

Reply:
Ohhh I thought it was just a condition that you pass in. I was wondering why you would pass in the currentValue outside of the reduce method because both accumulator + currentValue only exist in that method. So filterCriteria right now is just based upon a string, of either course(I should name it subject) or instructor. That way, the method knows what expression to use since all that data lives in the data.course array.